k8s/cmd/commands/sidecar/inject.go

```diff
@@ -0,0 +1,313 @@
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse the pv and pvc of backup: %v", err)
+		}
+		opts.VineyardConfig.SpillConfig.PersistentVolumeSpec = *spillPVSpec
+		opts.VineyardConfig.SpillConfig.PersistentVolumeClaimSpec = *spillPVCSpec
+	}
+
+	sidecar := &v1alpha1.Sidecar{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DefaultSidecarLabelValue,
+			Namespace: namespace,
+		},
+		Spec: *opts,
+	}
+	return sidecar, nil
+}
+
+// InjectSidecarConfig injects the sidecar config into the workload
+func InjectSidecarConfig(workloadObj, sidecarObj *unstructured.Unstructured) error {
+	template := workloadObj.Object["spec"].(map[string]interface{})["template"].(map[string]interface{})
+	// get the template spec of the workload
+	templateSpec := template["spec"].(map[string]interface{})
+	// get the labels of the workload
+	labels := template["metadata"].(map[string]interface{})["labels"].(map[string]interface{})
+	// get the containers of the workload
+	containers := templateSpec["containers"].([]interface{})
+	// get the volumes of the workload
+	if templateSpec["volumes"] == nil {
+		templateSpec["volumes"] = []interface{}{}
+	}
+	volumes := templateSpec["volumes"].([]interface{})
+
+	for i := range containers {
+		// wait for the sidecar container to be ready
+		command := containers[i].(map[string]interface{})["command"].([]interface{})
+		finalCmd := command[len(command)-1].(string)
+		finalCmd = "while [ ! -e /var/run/vineyard.sock ]; do sleep 1; done;" + finalCmd
+		command[len(command)-1] = fmt.Sprintf("%v", finalCmd)
+	}
+	mountPath := "/var/run"
+	pvcName, _, err := unstructured.NestedString(sidecarObj.Object, "spec", "volume", "pvcName")
+	if err != nil {
+		return fmt.Errorf("failed to get the mount path of the sidecar: %w", err)
+	}
+	if pvcName != "" {
+		mountPath, _, err = unstructured.NestedString(sidecarObj.Object, "spec", "volume", "mountPath")
+		if err != nil {
+			return fmt.Errorf("failed to get the mount path of the sidecar: %w", err)
+		}
+	}
+	// add volumeMounts to the containers
+	for i := range containers {
+		if containers[i].(map[string]interface{})["volumeMounts"] == nil {
+			containers[i].(map[string]interface{})["volumeMounts"] = []interface{}{}
+		}
+		containers[i].(map[string]interface{})["volumeMounts"] = append(
+			containers[i].(map[string]interface{})["volumeMounts"].([]interface{}),
+			map[string]interface{}{
+				"mountPath": mountPath,
+				"name":      "vineyard-socket",
+			})
+	}
+
+	// add labels to the workload for vineyard rpc service
+	labels[DefaultSidecarLabelName] = DefaultSidecarLabelValue
+
+	containers = append(containers, sidecarObj.Object["spec"].(map[string]interface{})["containers"].([]interface{})...)
+	volumes = append(volumes, sidecarObj.Object["spec"].(map[string]interface{})["volumes"].([]interface{})...)
+	templateSpec["containers"] = containers
+	templateSpec["volumes"] = volumes
+
+	return nil
+}
```

Comment:
Could you please take a try to improve the code reuse? Looks quite same with https://github.com/v6d-io/v6d/blob/dev%2Fkubernetes/k8s/pkg/webhook/sidecar/sidecar_webhook.go#L190

Why it cannot be deserailzed as and object (rather than plain yaml object) then reuse some existing logic?